### PR TITLE
[ClangImporter] Add support for `swift_attr(“@Sendable”)` on ObjC protocols

### PR DESCRIPTION
**Pull request by xedin**
When ClangImporter imports a nominal type with `@Sendable` attached,
it adds a `SynthesizedProtocolAttr` to the type. That did the right thing 
for concrete types, but protocols currently ignore these attributes. 

Manually add them where necessary.

Resolves: rdar://99758612

Partially based on https://github.com/apple/swift/pull/66188 (which I discovered later :))

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Review by slavapestov — APPROVED** [review]

File: lib/AST/RequirementMachine/RequirementLowering.cpp [hunk 1]
```diff
@@ -858,7 +858,7 @@ void swift::rewriting::expandDefaultRequirements(ASTContext &ctx,
 }
 
 /// Collect structural requirements written in the inheritance clause of an
-/// AssociatedTypeDecl or GenericTypeParamDecl.
+/// AssociatedTypeDecl, GenericTypeParamDecl, or ProtocolDecl.
```
Comment (slavapestov) [inline lib/AST/RequirementMachine/RequirementLowering.cpp:861]: Ah, oops :)

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Comment by xedin** [comment]
@swift-ci please smoke test macOS platform

**Pull request merged**